container/lxd/connection.go

```diff
@@ -141,7 +143,14 @@ func ConnectRemote(spec ServerSpec) (lxd.ContainerServer, error) {
 		return nil, errors.Trace(err)
 	}
 	client, err := lxd.ConnectLXD(uri, spec.connectionArgs)
-	return client, errors.Trace(err)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if spec.Project != "" {
+		client.UseProject(spec.Project)
```

Comment:
I got this to work with:
```go
client = client.UseProject(spec.Project)
```

Reply:
Doh!
That would explain that part :)

I'm still confused as to why I couldn't get the new config key to stick when adding a new cloud to a shared controller though. But maybe it's my testing that was wrong somehow.

Reply:
I did it with a fresh bootstrap, using `--config`, which adds it to model defaults.
`juju model-config` showed me the project I configured.
If you can give more detail on what you did I can try to chase it down...

Reply:
I had an existing juju controller which I upgraded to my local dev version through `juju upgrade-controller --build-agent`, I then created a local cloud YAML for my remote LXD cluster containing `project: foo` in the config section, then added that and matching credentials locally first and then to the existing controller.

This did show up as a new cloud on the controller but the `project` config key did not make it over.

(Sorry, that's from memory, I don't have that juju dev environment here anymore)